region/compressor.go

```diff
@@ -0,0 +1,106 @@
+// Copyright (C) 2020  The GoHBase Authors.  All rights reserved.
+// This file is part of GoHBase.
+// Use of this source code is governed by the Apache License 2.0
+// that can be found in the COPYING file.
+
+package region
+
+import (
+	"encoding/binary"
+	"fmt"
+	"net"
+
+	"github.com/tsuna/gohbase/compression"
+)
+
+type compressor struct {
+	compression.Codec
+}
+
+func min(x, y uint32) int {
+	if x > y {
+		return int(y)
+	}
+	return int(x)
+}
+
+func growBuffer(b []byte, sz int) []byte {
+	l := len(b) + sz
+	if l <= cap(b) {
+		return b[:l]
+	}
+	return append(b, make([]byte, sz)...)
+}
+
+func (c *compressor) compressCellblocks(cbs net.Buffers, uncompressedLen uint32) []byte {
+	b := newBuffer(4)
+
+	// put uncompressed length
+	binary.BigEndian.PutUint32(b, uncompressedLen)
+
+	uncompressedBuffer := newBuffer(min(uncompressedLen, c.ChunkLen()))
+	var chunkLen uint32
+	var lenOffset int
+	for {
+		n, err := cbs.Read(uncompressedBuffer)
+		if n == 0 {
+			break
+		}
+
+		// grow for chunk length
+		lenOffset = len(b)
+		b = growBuffer(b, 4)
+
+		b, chunkLen = c.Encode(uncompressedBuffer[:n], b)
+
+		// write the chunk length
+		binary.BigEndian.PutUint32(b[lenOffset:], chunkLen)
+
+		if err != nil {
```

Comment:
is any error expected here other than EOF?

Reply:
Done, panicing on any other error